src/commcare_cloud/fab/operations/formplayer.py

```diff
@@ -23,7 +23,8 @@ def _formplayer_jars_differ(build_dir, release_1, release_2):
     with cd(build_dir):
         with settings(warn_only=True):
             result = sudo("diff {}/libs/formplayer.jar {}/libs/formplayer.jar".format(release_1, release_2))
```

Comment:
is `diff` a good way to do this sort of thing? I thought it wasn't very good for binary files.

It might be better to see if we can upload the md5sum to s3 as well and we could download that first and not even download the jar at all

Reply:
I think diff is fine at it in the sense that it'll reliably have a return code of 1 if they differ and 0 if they don't. Have you heard/seen otherwise? Of course, it won't actually give a good text diff for binary files, but that's fine for our purposes.

Reply:
In terms of not needing to actually download the large file to get a diff, I agree that's a good thing to try.

Reply:
> I think diff is fine at it in the sense that it'll reliably have a return code of 1 if they differ and 0 if they don't.

Yeah I think it's fine from a "are these two files the same" standpoint. I haven't looked at actually diffing files recently, but I thought binary files were pretty bad for diff since they may have trouble finding anchor points for diff if there actually are changes. Though perhaps those algorithms are fairly sophisticated and can abort if it takes too long

so basically I think it's likely very fast for when these are the same, but there may be some set of differences where diff takes quite a while. I think comparing output from  `md5sum` would never have that issue

Reply:
I see, I hadn't thought of the performance on binary files. I'm used to it just saying "the binary files differ" which I assumed was easy and quick for it to do.

Anyway I think reworking to use md5 sums would be very reasonable, and something I might do.